File: kubedock/dns_management/plugins/example/entry.py

# KuberDock - is a platform that allows users to run applications using Docker
# container images and create SaaS / PaaS based on these applications.
# Copyright (C) 2017 Cloud Linux INC
#
# This file is part of KuberDock.
#
# KuberDock is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# KuberDock is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with KuberDock; if not, see <http://www.gnu.org/licenses/>.

"""Minimal example for DNS management plugin.
"""

from flask import current_app


def delete_type_A_record(domain, **kwargs):
    """
    Delete A record to domain

    :param domain: domain which will have been deleted
    :param dict kwargs: additional params such as address, token for access to
        external DNS management API
    :return: None

    Note: this is an example, does nothing, useful for testing
    """
    current_app.logger.debug(
        u'Called "delete_type_A_record" domain: {}, params: {}'.format(
            domain, kwargs))


def create_or_update_type_A_record(domain, new_ips, **kwargs):
    """
    Create or update A record for IP addresses of load
    balancer

    :param str domain: New subdomain name in existing zone
    :param list new_ips: IP addresses of load balancer (Ingress controller)
    :param dict kwargs: additional params such as address, token for access to
        external DNS management API
    :return: None

    Note: this is an example, does nothing, useful for testing
    """
    current_app.logger.debug(
        u'Called "create_or_update_type_A_record" domain: {}, ips: {}, '
        u'params: {}'.format(domain, new_ips, kwargs))

File: kubedock/frontend/static/js/app_data/entry.js
/*
 * KuberDock - is a platform that allows users to run applications using Docker
 * container images and create SaaS / PaaS based on these applications.
 * Copyright (C) 2017 Cloud Linux INC
 *
 * This file is part of KuberDock.
 *
 * KuberDock is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * KuberDock is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with KuberDock; if not, see <http://www.gnu.org/licenses/>.
 */

require('bootstrap/dist/js/bootstrap.min');
require('entry.less');

window.jQuery = window.$ = require('jquery');  // required for integration tests

const init = function(){
    if (!PROD){  // NOTE: UglifyJs will cut whole block if PROD is false
        // make those vars available in browser console for debugging
        window._ = _;
        window.Backbone = Backbone;

        if (/^https?:\/\/[^\/]+\/test(?:\?.*)?$/.test(window.location.href)){
            if (TEST){  // just run tests
                // for correct tracebacks in mocha
                require('stack-source-map')();
                const chai = require('chai');
                chai.use(require('chai-jquery'));
                chai.use(require('sinon-chai'));
                chai.use(require('chai-datetime'));

                // import tests using mocha-loader; run; exit
                require('mocha!app_data/tests/specs');
            } else {
                alert(  // eslint-disable-line no-alert
                    'You need to enable tests in config.js or build with ' +
                    '"ENABLE_TESTS=true"');
                window.location.href = '/';
            }
            return;
        }
    }

    var App = require('app_data/app');
    require('app_data/model');  // load all models and add resourcePremisers to App

    App.start();
};

init();
